cardinal/config.go

```diff
@@ -33,39 +48,104 @@ var defaultConfig = WorldConfig{
 type RunMode string
 
 type WorldConfig struct {
-	RedisAddress              string  `config:"REDIS_ADDRESS"`
-	RedisPassword             string  `config:"REDIS_PASSWORD"`
-	CardinalNamespace         string  `config:"CARDINAL_NAMESPACE"`
-	CardinalMode              RunMode `config:"CARDINAL_MODE"`
-	BaseShardSequencerAddress string  `config:"BASE_SHARD_SEQUENCER_ADDRESS"`
-	BaseShardQueryAddress     string  `config:"BASE_SHARD_QUERY_ADDRESS"`
-	CardinalLogLevel          string  `config:"CARDINAL_LOG_LEVEL"`
-	StatsdAddress             string  `config:"STATSD_ADDRESS"`
-	TraceAddress              string  `config:"TRACE_ADDRESS"`
+	// Cardinal
+	CardinalMode      RunMode `config:"CARDINAL_MODE"`
+	CardinalNamespace string  `config:"CARDINAL_NAMESPACE"`
+	CardinalLogLevel  string  `config:"CARDINAL_LOG_LEVEL"`
+
+	// Redis
+	RedisAddress  string `config:"REDIS_ADDRESS"`
+	RedisPassword string `config:"REDIS_PASSWORD"`
+
+	// Shard networking
+	BaseShardSequencerAddress string `config:"BASE_SHARD_SEQUENCER_ADDRESS"`
+	BaseShardQueryAddress     string `config:"BASE_SHARD_QUERY_ADDRESS"`
+
+	// Telemetry
+	StatsdAddress string `config:"STATSD_ADDRESS"`
+	TraceAddress  string `config:"TRACE_ADDRESS"`
+}
+
+func LoadWorldConfig() WorldConfig {
```

Comment:
It would be more idiomatic to have this return `(WorldConfig, error)`, instead of using log.Fatal to end early.